keymapper/injection/consumers/keycode_mapper.py

```diff
@@ -430,22 +433,41 @@ def handle_keycode(self, event, action, forward=True):
                 unreleased_entry = unreleased[type_and_code]
                 target_type, target_code = unreleased_entry.target_type_code
                 del unreleased[type_and_code]
-
+                
+                target_uinput = None
+                if isinstance(target_code, tuple):
+                    target_uinput = target_code[1]
+                    target_code = target_code[0]
```

Comment:
I think ideally it should be something like this, to keep the datastructure simple.

```py
target_uinput, target_type, target_code = unreleased_entry.target_type_code
```

and since this is a 3 tuple, `target_type_code` is not an appropriate name anymore. Maybe just `target`?

Reply:
> I think ideally it should be something like this, to keep the datastructure simple.

good idea, I just implemented it.
> The `keycode_mapper` was really hard to get at least somewhat readable.

Indeed, it is quite the challenge to see what is going on. I'm somewhat tempted to refactor the whole thing. Maybe breakout all the different logic paths into different functions and create a `event_handler_factory()` which will return the correct function. But not in this PR

Reply:
> I'm somewhat tempted to refactor the whole thing

understandable. It should probably be at least split into more functions. But yeah sure, lets do this in another PR.